docsrc/api.rst

```diff
@@ -11,11 +11,22 @@ Classes
 CmdStanModel
 ============
 
-.. autoclass:: cmdstanpy.CmdStanModel
-   :members:
+A CmdStanModel object encapsulates the Stan program. It manages program compilation and provides the following inference methods:
+
+sample
+    runs the HMC-NUTS sampler to produce a set of draws from the posterior distribution.
 
+optimize
+    produce a penalized maximum likelihood estimate (point estimate) of the model parameters.
```

Comment:
`summary` and `stanvariable` don't work with optimize. I guess the commandline summary doesn't work, so maybe that's a miss, but I think `stanvariable` should. This can be followup stuff.

Reply:
And I wasn't able to get variational running, but I think that interface should mimic the MCMC as much as possible.

Reply:
yes, followup PR will generalize this to all methods.
but wanted to proceed stepwise as much as possible.

Reply:
next PR, for sure.

Reply:
Sounds good